### PR TITLE
Error messages from a CLI should be sent to stderr

### PR DESCRIPTION
**Pull request by alikins**
##### ISSUE TYPE
- Bugfix Pull Request
##### ANSIBLE VERSION

```
ansible 2.1.0 (pr_10555_cobbler_auth_rebase 9278591758) last updated 2016/04/18 17:29:55 (GMT -400)
  lib/ansible/modules/core: (detached HEAD 5409ed1b28) last updated 2016/04/18 17:15:21 (GMT -400)
  lib/ansible/modules/extras: (detached HEAD 3afe117730) last updated 2016/04/18 17:15:21 (GMT -400)
  config file = /home/adrian/.ansible.cfg
  configured module search path = Default w/o overrides

<!--- Paste verbatim output from “ansible --version” between quotes -->
```
##### SUMMARY

This is a rebase of the pr at https://github.com/ansible/ansible/pull/10776
The original branch referenced in the pr is gone, so this is a rebase of that branch
pushed to my fork.

### CODE REVIEW

**Comment by alikins** [comment]
ping @romain-dartigues

File: contrib/inventory/windows_azure.py [hunk 1]
```diff
@@ -195,9 +195,7 @@ def add_cloud_services(self):
             for cloud_service in self.sms.list_hosted_services():
                 self.add_deployments(cloud_service)
         except WindowsAzureError as e:
-            print("Looks like Azure's API is down:")
-            print("")
-            print(e)
+            prin("Looks like Azure's API is down:")            print(e)
```
Comment (sivel) [inline contrib/inventory/windows_azure.py]: This line is generating syntax errors.  `prin` should likely be `print` and the second `print` should be on it's own line.

Comment (sivel) [inline contrib/inventory/windows_azure.py]: And based on what this PR is attempting to achieve, I think this should also go to stderr

File: contrib/inventory/linode.py [hunk 2]
```diff
@@ -185,19 +185,16 @@ def get_nodes(self):
...
 
     def get_node(self, linode_id):
         """Gets details about a specific node."""
         try:
             return Linode.find(api_id=linode_id)
         except chube_api.linode_api.ApiError as e:
-            print("Looks like Linode's API is down:")
-            print("")
-            print(e)
+            sys.stderr.write("Looks like Linode's API is down:\n")            sys.stderr.write(e + "\n")
```
Comment (sivel) [inline contrib/inventory/linode.py]: syntax error here too.

File: contrib/inventory/consul_io.py [hunk 3]
```diff
@@ -189,8 +190,8 @@ def setup_logging():
 try:
   import consul
 except ImportError as e:
-  print("""failed=True msg='python-consul required for this module. see
-  http://python-consul.readthedocs.org/en/latest/#installation'""")
+  sys.stderr.write("""failed=True msg='python-consul required for this module. see
+  http://python-consul.readthedocs.org/en/latest/#installation'\n""")
   sys.exit(1)
```
Comment (bcoca) [inline contrib/inventory/consul_io.py]: wouldn't it be better to?:

```
sys.exit("""failed=True msg='python-consul required for this module. seehttp://python-consul.readthedocs.org/en/latest/#installation'\n""")
```

^ will exit with rc=1 and print to stderr in py2/py3 compatible fashion.

**Comment by alikins** [comment]
Ooof. Fixing.

File: contrib/inventory/consul_io.py (hunk 3 above)
Comment (alikins) [inline contrib/inventory/consul_io.py]: Yup, especially if the only expected exit codes are 0/1.

**Comment by alikins** [comment]
Updated and forced push to branch.

File: contrib/inventory/digital_ocean.py [hunk 4]
```diff
@@ -147,7 +147,7 @@
 try:
     from dopy.manager import DoError, DoManager
 except ImportError as e:
-    print("failed=True msg='`dopy` library required for this script'")
+    sys.stderr.write("failed=True msg='`dopy` library required for this script'\n")
```
Comment (bcoca) [inline contrib/inventory/digital_ocean.py]: same change than in previous file, we only need sys.exit(" message" ), this applies to most changes below also, sorry I should have been more explicit with last comment i made.

**Comment by romain-dartigues** [comment]
@alikins sorry, for some reason I missed the notifications, I'll take a look at it ASAP :^\

**Pull request merged**